api/v1alpha1/virtualmachinepublishrequest_types.go

```diff
@@ -0,0 +1,530 @@
+	// kind equal to spec.target.location.kind, w/ the label
+	// "imageregistry.vmware.com/default".
+	//
+	// Please note that while optional, if a VirtualMachinePublishRequest sans
+	// target information is applied to a namespace without a default
+	// publication target, then the VirtualMachinePublishRequest resource
+	// will be marked in error.
+	//
+	// +optional
+	Target VirtualMachinePublishRequestTarget `json:"target,omitempty"`
+
+	// TTLSecondsAfterFinished is the time-to-live duration for how long this
+	// resource will be allowed to exist once the publication operation
+	// completes. After the TTL expires, the resource will be automatically
+	// deleted without the user having to take any direct action.
+	//
+	// If this field is unset then the request resource will not be
+	// automatically deleted. If this field is set to zero then the request
+	// resource is eligible for deletion immediately after it finishes.
+	//
+	// +optional
+	TTLSecondsAfterFinished *int64 `json:"ttlSecondsAfterFinished,omitempty"`
+}
+
+// VirtualMachinePublishRequestStatus defines the observed state of a
+//  VirtualMachinePublishRequest.
+type VirtualMachinePublishRequestStatus struct {
+	// CompletionTime represents time when the request was completed. It is not
+	// guaranteed to be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	//
+	// The value of this field should be equal to the value of the
+	// LastTransitionTime for the status condition Type=Complete.
+	//
+	// +optional
+	CompletionTime metav1.Time `json:"completionTime,omitempty"`
+
+	// StartTime represents time when the request was acknowledged by the
+	// controller. It is not guaranteed to be set in happens-before order
+	// across separate operations. It is represented in RFC3339 form and is
+	// in UTC.
+	//
+	// +optional
+	StartTime metav1.Time `json:"startTime,omitempty"`
+
+	// ImageName is the name of the VirtualMachineImage resource that is
+	// eventually realized in the same namespace as the VM and publication
+	// request after the publication operation completes.
+	//
+	// This field will not be set until the VirtualMachineImage resource
+	// is realized.
+	//
+	// +optional
+	ImageName string `json:"imageName,omitempty"`
+
+	// Ready is set to true only when the VM has been published successfully
+	// and the new VirtualMachineImage resource is ready.
+	//
+	// Readiness is determined by waiting until there is status condition
+	// Type=Complete and ensuring it and all other status conditions present
+	// have a Status=True. The conditions present will be:
+	//
+	//   * SourceValid
+	//   * TargetValid
+	//   * Uploaded
+	//   * ImageAvailable
+	//   * Complete
+	//
+	// +optional
+	Ready bool `json:"ready,omitempty"`
+
+	// Conditions is a list of the latest, available observations of the
+	// request's current state.
+	//
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
vm-operator-api has its own Condition type https://github.com/vmware-tanzu/vm-operator-api/blob/master/api/v1alpha1/condition_types.go#L45. Maybe we can use this?

Reply:
I'd like to move us to using the standard `metav1.Condition` type and improve on that one where necessary.

Reply:
I agree with @yi0909 that we will continue to use the existing Condition type for now.